terralab/version_utils.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import requests
+import os
+import json
+from datetime import datetime, date
+from importlib.metadata import version, PackageNotFoundError
+
+from terralab.config import load_config
+
+package_name = "terralab-cli"
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_version_info_file_path() -> str:
+    """Get the path to the version info file for storing the last version check date."""
+    config = load_config()
+    local_storage_dir = config.local_storage_path
+    os.makedirs(local_storage_dir, exist_ok=True)
+    return os.path.join(local_storage_dir, "version_info.json")
+
+
+def get_last_version_check_date() -> date | None:
+    """Get the date when the latest version was last checked."""
+    info_file = get_version_info_file_path()
+    try:
+        if os.path.exists(info_file):
+            with open(info_file, "r") as f:
+                data = json.load(f)
+                return datetime.strptime(
+                    data.get("last_version_check", ""), "%Y-%m-%d"
+                ).date()
+    except (ValueError, KeyError):
+        pass
```

Comment:
should we delete the file in case it got corrupted or something?

Reply:
that's a great idea, done in ad1d4c1f6d72d5a305792449104537743724ce25